packages/gatsby/src/redux/actions/public.js

```diff
@@ -364,9 +378,25 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     internalComponentName = `Component${pascalCase(page.path)}`
   }
 
+  const truncateLongPaths = store.getState().config.truncateLongPaths
+  const pagePath = truncateLongPaths ? truncatePath(page.path) : page.path
+
+  if (pagePath.length > 255) {
+    report.panic({
+      id: `11331`,
+      context: {
+        pluginName: name,
+        api: `createPages`,
+        message: chalk.bold.yellow(
+          `This path contains directory/file names that exceed OS character limit`
+        ),
+      },
+    })
+  }
```

Comment:
This error doesn't print problematic path(s). I think it would be beneficial for users to know about problematic paths so they are able to make custom truncation in their own code - the problematic paths will very likely from programmatic `createPage` call and not from creating pages from `/src/pages` dir (as those are files already, they don't violate file/dir name length limit).

Now that I think more about it:

I don't think we should try to modify page path in production builds ever (because it's unpredictable what page path will look like, the links that user expect to work, won't) so my proposition here would be this:

- don't add new config option (we should avoid those if there is flow that will be ok for all cases)
- depending if we are in gatsby build or develop (we can use ``process.env.NODE_ENV === `production` `` check to determine if we are in `gatsby build`) we would:
  - for `gatsby build`:
     - use `reporter.panic` and have a message that actually show the problematic path.
  - for `gatsby develop`:
     - use `truncatePath`
     - use `reporter.panicOnBuild` (or `reporter.error`) to show message (ideally same error id/code, but the text would be a bit different, because we don't want to terminate the session yet). Show problematic path, show what path this was automatically changed to. Mention that this will fail production build. This way user will get all the problematic paths

Reply:
Sounds reasonable. I'll make changes then.

Reply:
@pieh Done!